src/Engine.py

```diff
@@ -9,7 +9,8 @@
 
 class Engine:  
```

Comment:
As I have said, it will be great if you can have a note on what the function does and what the parameters of the function are.